src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs

```diff
@@ -308,7 +308,7 @@ private void EmitNativeToInterp(StreamWriter w, List<PInvokeCallback> callbacks)
 
             string module_symbol = _fixupSymbolName(method.DeclaringType!.Module!.Assembly!.GetName()!.Name!);
             uint token = (uint)method.MetadataToken;
-            string class_name = method.DeclaringType.Name;
+            string class_name = _fixupSymbolName(method.DeclaringType.Name);
             string method_name = method.Name;
             string entry_name = $"wasm_native_to_interp_{module_symbol}_{class_name}_{method_name}";
```

Comment:
Maybe the fixup should be done on `entry_name`, so we cover the whole symbol name.